driver/compile_common.ml

```diff
@@ -114,9 +114,23 @@ let implementation info ~backend =
     List.iter (fun suf -> remove_file (suf info)) sufs;
   in
   Misc.try_finally ?always:None ~exceptionally (fun () ->
-    let parsed = parse_impl info in
     if Clflags.(should_stop_after Compiler_pass.Parsing) then () else begin
-      let typed = typecheck_impl info parsed in
+      let typed =
+        match Filename.check_suffix info.source_file ".cmt" with
```

Comment:
Have you considered implementing this check alongside existing logic in `Compenv.process_action` for recognizing which pass to start from based on the input filename? You would need to add Lambda to Clflags.Compiler_pass.t.